From e2f43db3a88fa828c94ed4320915937d5b8247cc Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Thu, 1 Sep 2016 13:25:47 -0400
Subject: [PATCH 46/46] ARM: dts: k2g: disable dma-coherency for PCIe

The DMA coherency is broken for Keystone PCIe in the latest kernel . Add
an intermediate solution of removing the dma-coherency property in
the DT bindings to the driver functional for PCI network devices. Currently
because of this issue, the PCIe NIC cards doesn't function. This used to
work before. Even with this removed, we get reasonable performance with
intel e1000e PCI card comparable to other TI devices. So for now add this
intermediate solution before the coherency issue is root caused. This
appears to be a problem with the dma-coherency mask set incorrectly for
the PCI network device.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2g.dtsi | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/boot/dts/keystone-k2g.dtsi b/arch/arm/boot/dts/keystone-k2g.dtsi
index 13b964a..42d999a 100644
--- a/arch/arm/boot/dts/keystone-k2g.dtsi
+++ b/arch/arm/boot/dts/keystone-k2g.dtsi
@@ -883,7 +883,6 @@
 			status = "disabled";
 			device_type = "pci";
 			num-lanes = <1>;
-			dma-coherent;
 
 			/* error interrupt */
 			interrupts = <GIC_SPI 60 IRQ_TYPE_EDGE_RISING>;
-- 
1.9.1

